packages/flutter/lib/src/material/chip.dart

```diff
@@ -2037,7 +2037,12 @@ class _RenderChip extends RenderBox with SlottedContainerRenderObjectMixin<_Chip
     }
   }
 
+  final LayerHandle<OpacityLayer> _avatarOpacityLayerHandler = LayerHandle<OpacityLayer>();
+
   void _paintAvatar(PaintingContext context, Offset offset) {
+    final OpacityLayer? oldLayer = _avatarOpacityLayerHandler.layer;
+    _avatarOpacityLayerHandler.layer = null;
```

Comment:
This will likely cause problems. Assigning null to LayerHandler.layer reduces the ref count on the Layer and potentially disposes it if it goes to zero. If that happens, you'll get an exception on line 2057 for using a disposed object.

See also the [docs for this property](https://master-api.flutter.dev/flutter/rendering/LayerHandle/layer.html
):

> Setting a new value or null will dispose the previously held layer if there are no other open handles to that layer.

Like I said before: You only want to null this property out for branches where the layer isn't reused.

Reply:
yes!!! all tests passed